log_outgoing_requests/admin.py

```diff
@@ -45,4 +54,54 @@ def has_change_permission(self, request, obj=None):
     def query_params(self, obj):
         return obj.query_params
 
+    def change_view(self, request, object_id, extra_context=None):
+        """Decode request/response body and add to context for display"""
+
+        log = get_object_or_404(OutgoingRequestsLog, id=object_id)
+
+        log_req_body = decode(log.req_body, log.req_body_encoding)
+        log_res_body = decode(log.res_body, log.res_body_encoding)
```

Comment:
might be good to have a `try: except:` around each of these and catch the decode errors. If that happens, you can try decoding it again like [here](https://github.com/psf/requests/blob/main/requests/models.py#L932) where the `errors="replace"` is most relevant.

I can also live with doing the `errors="replace"` directly without try/except.

Do note the `LookupError` and `TypeError` handling there, I think this should also be covered.

Finally, I think it makes sense to have custom properties on the model for this instead of doing this in the view, that makes it easier to unit test those implementations *and* you can call the properties directly in the template (which is a bit more robust with unexpected exception handling), so something like:

```py
from django.utils.functional import cached_property

@cached_property
def decoded_request_body(self) -> str:
    try:
        decoded = str(self.req_body, self.req_body_encoding, errors="replace")
    except (LookupError, TypeError):
        decoded = str(self.req_body, errors="replace")
    return decoded
```